### PR TITLE
Simple(r) 2D detection debuggin overlay

### PR DESCRIPTION
**Pull request by Purg**
Add a C++ node to collect images and render received 2D detections on the corresponding specific source image if it is in our history.

### CODE REVIEW

File: ros/angel_utils/include/angel_utils/rate_tracker.hpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  /// Get the average tick rate from the window of measurements.
+  ///
+  /// If there have been no measurements taken yet (by calling `tick()`) then
+  /// a -1 value is returned.
+  ///
+  /// \return Average rate in Hz between tick measurements.
+  double get_rate_avg();
+
+private:
+  struct PImpl;
```
[the remark this answers is not in the document]
Comment (Purg) [reply ros/angel_utils/include/angel_utils/rate_tracker.hpp:40]: This is a programming pattern to hide implementations from the public interface that the user includes (see [here](https://en.cppreference.com/w/cpp/language/pimpl), [here](https://en.wikipedia.org/wiki/Opaque_pointer), and [here](https://docs.microsoft.com/en-us/cpp/cpp/pimpl-for-compile-time-encapsulation-modern-cpp?view=msvc-170) ). In this way, all implementation details not necessary to use the public functions and properties are hidden.

This reduces the complexity of reading the header for *how* to use it, as well as prevents the user from having to transitively include resources and dependencies that are not needed. In this case it is very simple, I'm at best hiding that we're using `std::chrono` for timing here, but in other cases more complicated things could be masked, like the use of OpenCV or Boost.

An advantage for larger libraries/packages is to effectively reducing compilation times by isolating changes to just their respective compilation units. For example, changing the `RateTracker` use something different than `std::chrono` will just effect the compilation of the object file for the implementation `rate_tracker.cpp` file, however the header stayed the same so anything that included the header does not need to be recompiled (re-linking would of course need to occur, but that's not avoidable and comparatively fast).

**Comment by Purg** [comment]
Added fixup commits. Let me know if they're OK by you and I can rebase + merge.

File: ros/angel_utils/include/angel_utils/rate_tracker.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshanderson-kw) [reply ros/angel_utils/include/angel_utils/rate_tracker.hpp:40]: Ok, that makes sense. Thanks for the links!

**Comment by joshanderson-kw** [comment]
> Added fixup commits. Let me know if they're OK by you and I can rebase + merge.

Looks good to me!

**Pull request merged**